config/routes.rb

```diff
@@ -106,14 +106,16 @@
 
   resources :communities, only: %i[show] do
     member do
-      get 'new_question'
+      get 'new_topic'
     end
   end
 
-  get 'answers/:id/versions', controller: "answers", action: "versions", as: "answer_version"
+  scope 'topics', controller: 'topics' do
+    get '/:id(/:title)/versions', action: 'versions', as: 'topic_versions'
+    get '/:id(/:title)', action: "show", as: 'topic'
+  end
 
-  get 'questions/:id(/:title)/versions', controller: "questions", action: "versions"
-  get 'questions/:id(/:title)', controller: "questions", action: "show", as: "question"
+  get 'posts/:id/versions', controller: "posts", action: "versions", as: "post_version"
```

Comment:
Let's also have a legacy route here to support old questions links for a while. Something like:

```ruby
get '/questions/:id(/:title)', action: 'show', controller: 'topics'
```

Reply:
@harigopal Just curious how would this be of help when the IDs are no way related? 🤔

Reply:
Oh. Silly me. @mahesh-krishnakumar Never mind.